packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -544,7 +544,7 @@ ${code}
         }
       }
 
-      if (!resolved || resolved === asset) {
+      if (!resolved) {
```

Comment:
Do you remember what case that second part was intended to prevent?

Reply:
Hmm, no I don't. But if all the tests pass... 😉